File: internship_cs_mar23/jury_solutions/cs/c.cs
using System;
using System.Linq;
using System.Threading;

namespace ConsoleApp2
{
	public class Program
	{
		private static void Solve()
		{
			var ints = ReadInts();
			var n = ints[0];
			var k = (long) ints[1];
			var a = ReadInts();
			var isZero = a.Select(e => e == 0 ? 1 : 0).ToArray();

			int finish = -1;
			int zeroesCount = 0;
			long sum = 0;
			long ans = 0;
			for (int start = 0; start < n; start++)
			{
				while (finish < n - 1 && a[finish + 1] + sum <= k && isZero[finish + 1] + zeroesCount <= 1)
				{
					sum += a[finish + 1];
					zeroesCount += isZero[finish + 1];
					finish++;
				}

				ans += finish - start + 1;
				sum -= a[start];
				zeroesCount -= isZero[start];

				if (finish < start)
				{
					sum += a[finish + 1];
					zeroesCount += isZero[finish + 1];
					finish++;
				}
			}

			Console.WriteLine(ans);
		}


		public static void Main()
		{
			var T = new Thread(Solve, 10000000);
			T.Start();
		}

		private static int[] ReadInts()
		{
			return Console.ReadLine()
				.Split()
				.Select(int.Parse)
				.ToArray();
		}
	}
}

File: internship_cs_mar23/jury_solutions/go/c.go
package main

import (
	"bufio"
	"fmt"
	"os"
)

func main() {
	reader := bufio.NewReader(os.Stdin)

	var n, k int
	fmt.Fscan(reader, &n)
	fmt.Fscan(reader, &k)

	a := make([]int, n)
	isZero := make([]int, n)

	for i := 0; i < n; i++ {
		fmt.Fscan(reader, &a[i])
		if a[i] == 0 {
			isZero[i] = 1
		} else {
			isZero[i] = 0
		}
	}

	finish := -1
	zeroesCount := 0
	sum := 0
	ans := 0
	for start := 0; start < n; start++ {
		for (finish < n-1) && (a[finish+1]+sum <= k) && (isZero[finish+1]+zeroesCount <= 1) {
			sum += a[finish+1]
			zeroesCount += isZero[finish+1]
			finish++
		}

		ans += finish - start + 1
		sum -= a[start]
		zeroesCount -= isZero[start]

		if finish < start {
			sum += a[finish+1]
			zeroesCount += isZero[finish+1]
			finish++
		}
	}

	fmt.Println(ans)
}
